### DISCUSSION TITLE
Image Attachments - please help :-)

### DISCUSSION THREAD

**Discussion by xnyjyh** [emailjs-com/help#158] [Q&A]
Im having some problems and dont what to add for my form to send properly after adding the "enctype="multipart/form-data"" into my form tag. I know I'm suppose to add some sort of script in to my existing script, but what and where do I add it? Please help to show me what to add and where, thanks! This is ALL my scripts for my form. More info... my form was sending before the Attachment was added.

Scripts -

<script type="text/javascript"
        src="https://cdn.jsdelivr.net/npm/@emailjs/browser@4/dist/email.min.js">
</script>

<script>
let dogCount = 1; // Tracks the number of dogs
const maxDogs = 4; // Maximum number of dogs allowed

// Add Dog Button Functionality
document.getElementById('addDogButton').addEventListener('click', () => {
    if (dogCount < maxDogs) {
        dogCount++;
        const dogDetailsContainer = document.getElementById('dogDetailsContainer');
        const newDogSection = `
            <div class="dog-details" id="dog${dogCount}">
                <button type="button" class="remove-dog" data-dog-id="dog${dogCount}">×</button>
                <label>Dog #${dogCount}'s Name:
                    <input type="text" name="dog${dogCount}Name" required>
                </label>
                <label>Is it safe for us to be in the yard with your dog?
                    <select name="dog${dogCount}Safety" required>
                        <option value="">Please Select</option>
                        <option value="Yes">Yes</option>
                        <option value="No">No</option>
                    </select>
                </label>
                <label>Additional comments about your dog:
                    <input type="text" name="dog${dogCount}Comments">
                </label>
            </div>
        `;
        dogDetailsContainer.insertAdjacentHTML('beforeend', newDogSection);

        // Disable button if max dogs reached
        if (dogCount === maxDogs) {
            document.getElementById('addDogButton').disabled = true;
        }

        // Attach remove functionality to the new close button
        attachRemoveFunctionality();
    }
});

// Attach Remove Functionality
function attachRemoveFunctionality() {
    document.querySelectorAll('.remove-dog').forEach(button => {
        button.onclick = (event) => {
            const dogId = event.target.getAttribute('data-dog-id');
            const dogElement = document.getElementById(dogId);
            if (dogElement) {
                dogElement.remove();
                dogCount--;
                document.getElementById('addDogButton').disabled = false; // Re-enable add button
            }
        };
    });
}

// Attach functionality to existing close buttons
attachRemoveFunctionality();
</script>

<button id="reserveButton" type="submit" class="full-width-button">Reserve Your Spot – $15 Deposit</button>

<script>
    // Initialize the slider background on page load
    function updateSliderBackground(slider) {
        const value = (slider.value - slider.min) / (slider.max - slider.min) * 100;
        slider.style.background = `linear-gradient(to right, #5BC0DE ${value}%, #D3D3D3 ${value}%)`;
    }

    function updateDogCountLabel(value) {
        document.querySelectorAll('.slider-labels span').forEach(el => el.style.fontWeight = 'normal');
        document.getElementById(`dogLabel${value}`).style.fontWeight = 'bold';
    }

    function updateFrequencyLabel(value) {
        document.querySelectorAll('.slider-labels span').forEach(el => el.style.fontWeight = 'normal');
        document.getElementById(`freqLabel${value}`).style.fontWeight = 'bold';
    }

    function calculatePrice() {
        const dogCount = parseInt(document.getElementById('dogCountSlider').value);
        const frequencyValue = parseInt(document.getElementById('frequencySlider').value);
        const frequencies = { 1: 'twice', 2: 'weekly', 3: 'biweekly', 4: 'monthly' };

        const prices = {
            1: { twice: 16, weekly: 20, biweekly: 30, monthly: 70 },
            2: { twice: 20, weekly: 24, biweekly: 34, monthly: 80 },
            3: { twice: 24, weekly: 28, biweekly: 38, monthly: 90 },
            4: { twice: 28, weekly: 32, biweekly: 42, monthly: 100 },
        };

        const frequency = frequencies[frequencyValue];
        let totalPrice = prices[dogCount][frequency];

        document.getElementById('totalPrice').innerText = totalPrice;
    }

    function showPriceAndForm() {
        calculatePrice();
        document.getElementById('priceContainer').style.display = 'block';
        document.getElementById('signupFormContainer').style.display = 'block';
    }

    // Initialize slider backgrounds on page load
    document.querySelectorAll('input[type="range"]').forEach(slider => {
        updateSliderBackground(slider);
    });
</script>

<script>
   document.getElementById('contact-form').addEventListener('submit', function(event) {
       event.preventDefault(); // Prevent default form submission behavior

       // Hide the button and show the loading message
       document.getElementById('reserveButton').style.display = 'none';
       document.getElementById('loadingMessage').style.display = 'block';

       // Your specific service and template IDs
       const serviceID = 'service_dum6gqe'; // Replace with your actual EmailJS service ID
       const templateID = 'template_vyogkou'; // Replace with your actual EmailJS template ID

       emailjs.sendForm(serviceID, templateID, this)
           .then(function(response) {
               console.log('SUCCESS!', response.status, response.text);
               // Redirect to Square payment link after successful submission
               window.location.href = 'https://square.link/u/njqWcF4m'; // Replace with your actual Square link
           }, function(error) {
               console.error('FAILED...', error);
               alert('FAILED... Please try again later.');
               // Revert to show the original button if submission fails
               document.getElementById('reserveButton').style.display = 'block';
               document.getElementById('loadingMessage').style.display = 'none';
           });
   });
</script>

**Comment by xr0master** [comment]
please fix the message
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

**Reply by xnyjyh** [reply]
What does that mean? that link shows nothing of what my question/concern was about.

**Reply by xnyjyh** [reply]
I just need to know what to put where so the image attachments works and sends. I get this when testing to send my form "FAILED... Please try again later." The tut on emailjs about attachments is very vague for someone like me who knows nothing of coding and scripts. Help would be very appreciated! :-) Thanks

**Reply by xnyjyh** [reply]
Why does this group even exist if no one helps? I don't understand. All I get is 'fix the message,' and I don't even know what that means. WTF!? who is this Maintainor? xr0master, master of what? I just need help, and been almost 5 days for nothing since this post. Help, please.

**Reply by xr0master** [reply]
Your post is difficult to follow because it’s unstructured and unformatted. If you’re looking for help, please take the time [to format your message properly](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) and include relevant details like the error, the context, and what you’ve already tried.

Remember, this is a community of volunteers. No one is obligated to assist, especially when the approach comes across as demanding or disrespectful.

If you’d like constructive help, kindly rephrase your query with the necessary details. That way, it’s easier for others to step in and support you.

**Reply by xnyjyh** [reply]
rephrase which? all the necessary info is in the original post though. I'm not a posting pro; I just post the core essentials to get an answer. Im very sure its pretty straight forward to what I'm looking for and what I posted. ok, so where can I get a real answer form then? I know the answer is very easy to you guys, but not to me. instructions on the website about attachments is fine for people who know things about scripts etc etc, not ALL people speak this language. I know everything about concrete and cooking, but not coding... And if someone asked and has many of times in my 27 years, id simple answer them. so I'm not too sure why its hard for you to help and answer. I already know you know the answer and I'm sorry if you have OCD about how I copy-paste my my script, but that's the script. I don't know what else you want me to do?!...

**Reply by xr0master** [reply]
I see.

1. Attachments are a paid feature, please tell me what subscription plan you have.
2. In the template, you also need to add a variable for attachments, did you add it?
